packages/opentelemetry-metrics/src/BatchObserverMetric.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import { Resource } from '@opentelemetry/resources';
+import { BatchObserverResult } from './BatchObserverResult';
+import { BoundObserver } from './BoundInstrument';
+import { Batcher } from './export/Batcher';
+import { MetricKind, MetricRecord } from './export/types';
+import { Metric } from './Metric';
+
+const NOOP_CALLBACK = () => {};
+const MAX_TIMEOUT_UPDATE_MS = 500;
+
+/** This is a SDK implementation of Batch Observer Metric. */
+export class BatchObserverMetric extends Metric<BoundObserver>
+  implements api.BatchObserver {
+  private _callback: (observerResult: api.BatchObserverResult) => void;
+  private _maxTimeoutUpdateMS: number;
+
+  constructor(
+    name: string,
+    options: api.BatchMetricOptions,
+    private readonly _batcher: Batcher,
+    resource: Resource,
+    callback?: (observerResult: api.BatchObserverResult) => void
+  ) {
+    super(name, options, MetricKind.OBSERVER, resource);
```

Comment:
As per [this comment](https://github.com/obecny/opentelemetry-js/blob/batch-observer/packages/opentelemetry-metrics/src/export/types.ts#L26), `MetricKind.OBSERVER` is being deprecated/removed. Why we are using it here, perhaps I am missing something? Why don't we use actually type instead of some deprecated type?

Reply:
when PR was created it was still needed not to have too many things in one PR already